src/la/linsys_PETSC.hh

```diff
@@ -91,6 +91,7 @@ public:
     PetscErrorCode mat_zero_entries() override
     {
         matrix_changed_ = true;
+        constraints_.clear();
```

Comment:
Rather use LocalSystem for assembly and local elimination of Dirichlet conditions.